bsip-0043.md

```diff
@@ -0,0 +1,96 @@
+
+However, there might be another opportunity to promote the asset and stimulate the trading - use native Bitshares referral program. At this time unfortunately an assets owner is not able to share market fees with registrars and referrers to stimulate them to market the asset trading, so we suggest to add this possibility. Furthermore, enabling this feature for MPAs (e.g. bitCNY or bitUSD) can provide additional bounty for Bitshares registrars and referrals which can lead to more traders joining to the ecosystem.
+The workflow may be as following:
+
+An asset issuer defines the market_fee_reward_percent - what percentage of the market fee he wants to share with the registrar in asset options. Registrar defines reward_percent for the referrer for each user(using the already existing BitShares mechanism).
+Market fee reward is accumulated on the user account. The user decides when they want to claim the market fee reward and move it to their active balance. There is another operation created asset_claim_reward_operation. Each user pays network fee to call this operation.
+
+# Motivation
+
+To make promoting BitShares and bringing new users much more attractive to registrars and referrers by sharing UIAs and MPAs market fees with them.
+
+# Rationale
+When 'fill_order_operation' executing at the moment of market fee calculation there will be calculate the reward for the registrar used the parameter market_fee_reward_percent. Then this market_fee_reward_percent of the market fee is split between the registrar and referrer according to the referrer_rewards_percentage, which is set up during the new account registration by registrar (please see the parameters for register_account).
+
+# Specifications
+
+## market_fee_reward_percent asset option
+Percent of market fee that will be paid to buyer's registrar. Set by UIA issuer.
+
+## market_sharing_whitelist asset option
+An optional list of accounts (configurable by the UIA Issuer) could provide increased control over who is eligible to get market rewards.
+
+## New database and wallet api methods
+**get_account_reward(account, asset_symbol)** - Returns information about the reward amount for specific asset.
+**list_account_rewards(account)** - Returns information about the reward amount in various assets
+**claim_reward(account, asset_symbol, amount)** - Claim account's reward for the given asset.
+
+## account_reward_object
+A new BitShares object type in implementation space impl_account_reward_object_type = 2.18.X that tracks the rewards of a single account/asset pair
+```
+account_reward_object {
+owner_account,
+asset_type,
+reward_amount
+}
+```
+## account_reward_index
+A new index that stores objects of account_reward_object-type in graphene::database and allow random fast access to objects by given criteria
+```
+account_reward_index multi-index of account_reward_objects
+indexed_by
+[id]
+[owner_account, asset_type]
+[asset_type, reward_amount desc, owner_account]
```

Comment:
Please consider moving indices that are not generally useful or consensus-related into optional plugins.

Reply:
We do not think that these indices belong to a plugin, because it's definitely consensus-related.

Reply:
> TBH I think this is over-detailed to be in a BSIP document. It's implementation details, which can be decided by the developer.

Agreed

Reply:
Perhaps you're right! We have already half-implemented this BSIP when we did a prototype as part of the investigation if it could be done at all. 
So this is already described and to take it out would mean to spend more time :)

Reply:
The BSIP should not tie you down, and allow to switch if anything would need to be restructured.

Put an abstract representation of what you did in it, not the actual representation.